### PR TITLE
Enhancements to RTEMS Driver  

### PR DESCRIPTION
**Pull request by heshamelmatary**

### CODE REVIEW

File: clang/lib/Driver/ToolChains/RTEMS.cpp [hunk 1]
```diff
@@ -225,6 +223,11 @@ void RTEMS::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
     SmallString<128> P(D.SysRoot);
     llvm::sys::path::append(P, "include");
     addExternCSystemInclude(DriverArgs, CC1Args, P.str());
+  } else {
+    // Add a default RTEMS target directory to the include path
+    SmallString<128> P(D.InstalledDir+"/../"+getTripleString());
```
Comment (arichardson) [inline clang/lib/Driver/ToolChains/RTEMS.cpp]: I would run `git-clang-format -f HEAD` to get consistent formatting.

Comment (heshamelmatary) [reply clang/lib/Driver/ToolChains/RTEMS.cpp]: Done

**Comment by heshamelmatary** [comment]
@arichardson Do we need to remove that? https://github.com/CTSRD-CHERI/llvm-project/blob/f6f0803392da64b428a047b4baac73882a35891b/clang/test/Driver/cheri/rtems.c#L13

**Comment by arichardson** [comment]
> @arichardson Do we need to remove that?
> 
> https://github.com/CTSRD-CHERI/llvm-project/blob/f6f0803392da64b428a047b4baac73882a35891b/clang/test/Driver/cheri/rtems.c#L13

What happens if you run `$BUILD_DIR/bin/llvm-lit -vv $SOURCE_DIR/clang/test/Driver/cheri/rtems.c`? If it fails then the file needs changing.

**Comment by heshamelmatary** [comment]
```
llvm-lit: /local/scratch/hmka2/Coding/RISCV/cheri-riscv-releases/llvm-project/llvm/utils/lit/lit/llvm/config.py:317: note: Running tests for CHERI_CAP_SIZE=16
llvm-lit: /local/scratch/hmka2/Coding/RISCV/cheri-riscv-releases/llvm-project/llvm/utils/lit/lit/llvm/config.py:405: note: using clang: /local/scratch/hmka2/Coding/RISCV/cheri-riscv-releases/llvm-project/build/bin/clang
-- Testing: 1 tests, 1 workers --
PASS: Clang :: Driver/cheri/rtems.c (1 of 1)

Testing Time: 0.17s
  Expected Passes    : 1

```

**Pull request merged**